### PR TITLE
Add types to private and final methods

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179 
| License       | MIT
| Doc PR        | N/A

Backports from #33154.

### CODE REVIEW

**Review by oliverquynh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
A mock in Workflow might have an issue, see tests.

**Comment by nicolas-grekas** [comment]
Maybe the fix related to the failing tests should go to #33197 btw?

**Comment by derrabus** [comment]
The test was right. I cannot type-hint for `WorkflowSupportStrategyInterface` yet because we also have to support a legacy interface there.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 1]
```diff
@@ -181,7 +181,7 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
         }
     }
 
-    private static function throwInvalidArgumentException($message, $trace, $i, $propertyPath)
+    private static function throwInvalidArgumentException(string $message, array $trace, int $i, $propertyPath): void
```
Comment (xabbuh) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: `string` for `$propertyPath` too?

Comment (derrabus) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: `$propertyPath` can be an object, IIRC.

Comment (derrabus) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: It worked. 🤷‍♂

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**